nncf/data/dataset.py

```diff
@@ -42,12 +44,12 @@ class Dataset(Generic[DataItem, ModelInput]):
     """
 
     def __init__(
-        self, data_source: Iterable[DataItem], transform_func: Optional[Callable[[DataItem], ModelInput]] = None
+        self, data_source: Sequence[DataItem], transform_func: Optional[Callable[[DataItem], ModelInput]] = None
```

Comment:
NNCF does not require `__len__` to be implemented in `data_source`. Please return back this change.